src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public FigureDrawable getRandomFigure() {
+        int value = random.nextInt(4);
+        String color = colorSupplier.getRandomColor();
+
+        switch (value) {
+            case 0: {
+                return new IsoscelesTrapezoid(color, random.nextInt(10) + 1,
+                        random.nextInt(10) + 1, random.nextInt(10) + 1);
+            }
+            case 1: {
+                return new Circle(color, random.nextInt(10) + 1);
+            }
+            case 2: {
+                return new RightTriangle(color, random.nextInt(10) + 1,
+                        random.nextInt(10) + 1);
+            }
+            case 3: {
+                return new Rectangle(color, random.nextInt(10) + 1,
+                        random.nextInt(10) + 1);
+            }
+            case 4: {
+                return new Square(color, random.nextInt(10) + 1);
```

Comment:
The random.nextInt(4) call will never result in the value 4, so this case will never be executed. You should adjust the range of random.nextInt() to include the number of cases you have.